src/wbi/experiment.py

```diff
@@ -125,160 +124,59 @@ def generate_cameraframedata_files(self, output_folder=None):
-
         stdev_brightness = np.std(adjusted_brightness)
-        (repeated_flash_loc,) = np.where(
+
+        (flash_loc,) = np.where(
             adjusted_brightness > stdev_brightness * config.flash_finder.std_factor
         )
 
-        if not repeated_flash_loc:
-            flash_loc = np.array([])
-        else:
-            flash_loc = [repeated_flash_loc[0]]
-            for index, flash_frame in enumerate(repeated_flash_loc, start=1):
-                if flash_frame != repeated_flash_loc[index - 1] + 1:
-                    flash_loc.append(flash_frame)
-            flash_loc = np.array(flash_loc)
-
-        x_pos, y_pos, z_pos = self.get_frame_positions()
-        volume_index = self.get_frame_timings()
+        # Remove flash frames that are consecutive and thus represent the same flash
+        flash_loc = flash_loc[np.nonzero(np.diff(flash_loc, prepend=-np.inf) != 1)]
+
+        data = self.timing_dataframe
+        # All the data we wish to save in a .mat file
+        mat_data = {
+            "imageIdx": data["frame_index"].to_numpy()[:, np.newaxis],
+            "frameTime": data["time"].to_numpy()[:, np.newaxis],
+            "flashLoc": (flash_loc + 1)[:, np.newaxis],  # 1-indexed
+            # TODO: Legacy code is taking a diff without a prepend, thus ends up 1 shorter than all the rest
+            # of the arrays here. We take values from 1: to emulate legacy behavior
+            "stackIdx": (data["volume_index"].to_numpy() + 1)[1:, np.newaxis],
+            # 1-indexed
+            "imSTD": stdev[:, np.newaxis],
+            "imAvg": brightness[:, np.newaxis],
+            "xPos": data["ludl_x"].to_numpy()[:, np.newaxis],
+            "yPos": data["ludl_y"].to_numpy()[:, np.newaxis],
+            "Z": data["piezo_position"].to_numpy()[:, np.newaxis],
+        }
 
         savemat(
             os.path.join(output_folder, "hiResData.mat"),
-            {
-                "dataAll": {
-                    "imageIdx": timing.timing[
-                        "frame_index"
-                    ].values.reshape(  # type:ignore
-                        timing.timing["frame_index"].shape[0], 1
-                    ),
-                    "frameTime": timing.timing["time"].values.reshape(  # type:ignore
-                        timing.timing["time"].shape[0], 1
-                    ),
-                    "flashLoc": (flash_loc + 1).reshape(flash_loc.shape[0], 1),
-                    "stackIdx": (volume_index + 1).reshape(volume_index.shape[0], 1),
-                    "imSTD": stdev.reshape(stdev.shape[0], 1),
-                    "imAvg": brightness.reshape(brightness.shape[0], 1),
-                    "xPos": x_pos.reshape(x_pos.shape[0], 1),
-                    "yPos": y_pos.reshape(y_pos.shape[0], 1),
-                    "Z": z_pos.reshape(z_pos.shape[0], 1),
-                }
-            },
+            mat_data,
         )
 
+        # TODO: The following file is generated for legacy reasons
```

Comment:
For all of these `TODO`s, I'll add relevant issues so we remember to revisit these in the future.